File: src/main/java/com/examportal/dto/StudentQueryDTO.java
package com.examportal.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class StudentQueryDTO {
    
    @NotBlank(message = "Title must be provided")
    private String title;
    
    @NotBlank(message = "Subject must be provided")
    private String subject;
    
    @NotBlank(message = "Query must be provided")
    private String query;
}

File: Dotnet3/ExamPortal/DTOs/StudentQueryDTO.cs
using System.ComponentModel.DataAnnotations;

namespace ExamPortal.DTOs
{
    public class StudentQueryDTO
    {
        [Required(ErrorMessage = "Title must be provided")]
        public string Title { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Subject must be provided")]
        public string Subject { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Query must be provided")]
        public string Query { get; set; } = string.Empty;
    }
}
